src/components/IEEEFunctionInput.vue

```diff
@@ -0,0 +1,108 @@
+<template>
+  <div class="function-input">
+    <b>𝔽(</b>
+    <small>
+        base:<input type="number" v-model="base" size="2" v-bind="$attrs" />,
+        mantissaSize:<input type="number" v-model="mantissaSize" size="5" v-bind="$attrs" />,
+        eMin:<input type="number" v-model="eMin" size="5" v-bind="$attrs" />,
+        eMax:<input type="number" v-model="eMax" size="5" v-bind="$attrs" />,
+        denorm:<input type="text" v-model="denorm" size="5" v-bind="$attrs" />
+    </small>
+    <b>)</b>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref, shallowRef, watch } from "vue";
+
+export default defineComponent({
+  props: {
+    mantissaSize: {
+      type: Number,
+      required: true,
+    },
+    base: {
+      type: Number,
+      required: true,
+    },
+    eMin: {
+      type: Number,
+      required: true,
+    },
+    eMax: {
+      type: Number,
+      required: true,
+    },
+    denormValue: {
+        type: String,
+        required: true,
+    },
+  },
+  emits: {
+    validFunction: (value: Boolean) => true,
```

Comment:
I recommend just returning the entire format here if it's valid.
And if it's invalid, return `null`.

Define this in some shared file
```ts
type IEEEFormat = {
  /**
   * Which base to use (binary, decimal, ...)
   */
  base: number;

  /**
   * Minimum exponent
   */
  eMin: number;

  /**
   * Maximum exponent
   */
  eMax: number;

  /**
   * Size of the mantissa in bits
   * (Does this include the implicit bit?)
   */
  mantissaSize: number;

  /**
   * If denormalized numbers should also be handled
   */
  allowDenormalized: boolean;
}
```

and then
```ts
validFunction: (value: IEEEFormat | null) => void,
```